File: frontend/src/modules/admin/types/adminStats.ts
/**
 * 管理员统计相关类型定义
 */

// =============================================================================
// 趋势数据
// =============================================================================

export interface TrendData {
  /** 用户数变化百分比 */
  users_change: number;
  /** 学生数变化百分比 */
  students_change: number;
  /** 教师数变化百分比 */
  teachers_change: number;
  /** 活跃率变化百分比 */
  active_rate_change: number;
}

// =============================================================================
// 概览统计
// =============================================================================

export interface OverviewStats {
  /** 总用户数 */
  total_users: number;
  /** 学生数量 */
  student_count: number;
  /** 教师数量 */
  teacher_count: number;
  /** 管理员数量 */
  admin_count: number;
  /** 今日活跃用户数 */
  active_users_today: number;
  /** 活跃率（百分比） */
  active_rate: number;
  /** 趋势数据 */
  trends: TrendData;
}

// =============================================================================
// 用户增长
// =============================================================================

export interface UserGrowthDataPoint {
  /** 日期 (YYYY-MM-DD) */
  date: string;
  /** 累计总用户数 */
  total: number;
  /** 累计学生数 */
  students: number;
  /** 累计教师数 */
  teachers: number;
}

export interface UserGrowthSummary {
  /** 期间新增用户总数 */
  total_new_users: number;
  /** 日均增长数 */
  avg_daily_growth: number;
}

export interface UserGrowthResponse {
  /** 统计周期 (7d/30d/90d) */
  period: string;
  /** 增长数据点列表 */
  data: UserGrowthDataPoint[];
  /** 增长摘要 */
  summary: UserGrowthSummary;
}

// =============================================================================
// 最近活动
// =============================================================================

export interface ActivityItem {
  /** 活动 ID */
  id: string;
  /** 用户名 */
  user_name: string;
  /** 操作描述 */
  action_display: string;
  /** 时间戳 */
  timestamp: string;
  /** 活动类型 (success/info/warning) */
  type: 'success' | 'info' | 'warning';
}

export interface RecentActivitiesResponse {
  /** 活动列表 */
  items: ActivityItem[];
  /** 总数 */
  total: number;
}

// =============================================================================
// 系统状态
// =============================================================================

export interface ServiceStatus {
  /** 服务名称 */
  name: string;
  /** 状态 (running/stopped/warning) */
  status: 'running' | 'stopped' | 'warning';
  /** 延迟（毫秒） */
  latency_ms: number | null;
}

export interface SystemAlert {
  /** 警告 ID */
  id: string;
  /** 标题 */
  title: string;
  /** 描述 */
  description: string;
  /** 严重程度 (error/warning/info) */
  severity: 'error' | 'warning' | 'info';
}

export interface SystemStatusResponse {
  /** 服务状态列表 */
  services: ServiceStatus[];
  /** 系统警告列表 */
  alerts: SystemAlert[];
}

// =============================================================================
// 用户增长周期类型
// =============================================================================

export type UserGrowthPeriod = '7d' | '30d' | '90d';

File: backend/app/api/v1/schemas/admin_stats.py
"""
管理员统计 API Schema 定义
"""

from datetime import datetime

from pydantic import BaseModel, Field

# =============================================================================
# 趋势数据
# =============================================================================


class TrendData(BaseModel):
    """趋势数据"""

    users_change: float = Field(description="用户数变化百分比")
    students_change: float = Field(description="学生数变化百分比")
    teachers_change: float = Field(description="教师数变化百分比")
    active_rate_change: float = Field(description="活跃率变化百分比")


# =============================================================================
# 概览统计
# =============================================================================


class OverviewStatsResponse(BaseModel):
    """概览统计响应"""

    total_users: int = Field(description="总用户数")
    student_count: int = Field(description="学生数量")
    teacher_count: int = Field(description="教师数量")
    admin_count: int = Field(description="管理员数量")
    active_users_today: int = Field(description="今日活跃用户数")
    active_rate: float = Field(description="活跃率（百分比）")
    trends: TrendData = Field(description="趋势数据")


# =============================================================================
# 用户增长
# =============================================================================


class UserGrowthDataPoint(BaseModel):
    """用户增长数据点"""

    date: str = Field(description="日期 (YYYY-MM-DD)")
    total: int = Field(description="累计总用户数")
    students: int = Field(description="累计学生数")
    teachers: int = Field(description="累计教师数")


class UserGrowthSummary(BaseModel):
    """用户增长摘要"""

    total_new_users: int = Field(description="期间新增用户总数")
    avg_daily_growth: float = Field(description="日均增长数")


class UserGrowthResponse(BaseModel):
    """用户增长响应"""

    period: str = Field(description="统计周期 (7d/30d/90d)")
    data: list[UserGrowthDataPoint] = Field(description="增长数据点列表")
    summary: UserGrowthSummary = Field(description="增长摘要")


# =============================================================================
# 最近活动
# =============================================================================


class ActivityItem(BaseModel):
    """活动项"""

    id: str = Field(description="活动 ID")
    user_name: str = Field(description="用户名")
    action_display: str = Field(description="操作描述")
    timestamp: datetime = Field(description="时间戳")
    type: str = Field(description="活动类型 (success/info/warning)")


class RecentActivitiesResponse(BaseModel):
    """最近活动响应"""

    items: list[ActivityItem] = Field(description="活动列表")
    total: int = Field(description="总数")


# =============================================================================
# 系统状态
# =============================================================================


class ServiceStatus(BaseModel):
    """服务状态"""

    name: str = Field(description="服务名称")
    status: str = Field(description="状态 (running/stopped/warning)")
    latency_ms: float | None = Field(default=None, description="延迟（毫秒）")


class SystemAlert(BaseModel):
    """系统警告"""

    id: str = Field(description="警告 ID")
    title: str = Field(description="标题")
    description: str = Field(description="描述")
    severity: str = Field(description="严重程度 (error/warning/info)")


class SystemStatusResponse(BaseModel):
    """系统状态响应"""

    services: list[ServiceStatus] = Field(description="服务状态列表")
    alerts: list[SystemAlert] = Field(description="系统警告列表")
